packages/ra-core/src/controller/CreateController.tsx

```diff
@@ -1,13 +1,44 @@
-import { Component } from 'react';
-import PropTypes from 'prop-types';
+import { Component, ReactNode } from 'react';
 import { connect } from 'react-redux';
 import compose from 'recompose/compose';
 import inflection from 'inflection';
 import { parse } from 'query-string';
 
-import translate from '../i18n/translate';
+import withTranslate from '../i18n/translate';
 import { crudCreate as crudCreateAction } from '../actions';
 import checkMinimumRequiredProps from './checkMinimumRequiredProps';
+import { Location } from 'history';
+import { match as Match } from 'react-router';
+import { Record, Translate, Dispatch } from '../types';
+import { RedirectionSideEffect } from '../sideEffect';
+
+interface ChildrenFuncParams {
+    isLoading: boolean;
+    defaultTitle: string;
+    save: (record: Partial<Record>, redirect: RedirectionSideEffect) => void;
+    resource: string;
+    basePath: string;
+    record?: Record;
+    redirect: RedirectionSideEffect;
+    translate: Translate;
+}
+
+interface Props {
+    basePath: string;
+    children: (params: ChildrenFuncParams) => ReactNode;
+    crudCreate: Dispatch<typeof crudCreateAction>;
+    hasCreate: boolean;
+    hasEdit: boolean;
+    hasList: boolean;
+    hasShow: boolean;
+    isLoading: boolean;
+    location: Location;
+    match: Match;
+    record?: Record;
+    resource: string;
+    title: string | ReactNode;
```

Comment:
You didn't transpose the propTypes correctly. `title`, for instance, was not required. Please double check for all components.

Reply:
Yes and actually, that's a prop of the `ra-ui-materialui` `Create` component.

Reply:
I saw you remove the title prop here, but I suspect you didn't cross check the isRequired attribute on all props. hasEdit, for instance, is required in your version but was not require din propTypes